src/222/main/kotlin/org/rust/cargo/project/model/impl/CargoExternalSystemProjectAware.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.model.impl
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.externalSystem.autoimport.*
+import com.intellij.openapi.externalSystem.autoimport.ExternalSystemSettingsFilesModificationContext.ReloadStatus
+import com.intellij.openapi.externalSystem.model.ProjectSystemId
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.util.PathUtil
+import com.intellij.util.containers.DisposableWrapperList
+import org.rust.cargo.CargoConstants
+import org.rust.cargo.project.model.CargoProjectsService
+import org.rust.cargo.project.model.CargoProjectsService.CargoRefreshStatus
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.model.impl.CargoSettingsFilesService.SettingFileType
+
+@Suppress("UnstableApiUsage")
+class CargoExternalSystemProjectAware(
+    private val project: Project
+) : ExternalSystemProjectAware {
+
+    private val listeners = DisposableWrapperList<ExternalSystemProjectListener>()
```

Comment:
Why do we need this list?

Reply:
Actually, we don't need it now
It's a part of my experiments during the implementation that I missed/forgot to remove